lib/SILOptimizer/SILCombiner/SILCombinerApplyVisitors.cpp

```diff
@@ -566,27 +566,62 @@ SILCombiner::optimizeConcatenationOfStringLiterals(ApplyInst *AI) {
   return tryToConcatenateStrings(AI, Builder);
 }
 
+// TODO:
+// Move this concrete type propagation for existentials out of sil-combine
+// We can then use DominanceAnalysis which is the right thing to do here
+// We avoid using it here due to compilation speed implications
+static bool fastDominates(SILInstruction *a, SILInstruction *b) {
+  auto aBlock = a->getParent(), bBlock = b->getParent();
+
+  // If the blocks are different, bail
+  if (aBlock != bBlock)
+    return true;
+
```

Comment:
It would be nicer if you could extract this code into a helper function (e.g. a static function of DominanceInfo) and not copy it from DominanceInfo

Reply:
On it - will extract the same block check into a helper function.